### PR TITLE
REL-2879: Added warning for invalid phone numbers

### PR DESCRIPTION
**Pull request by sraaphorst**
A new task came in requesting that the PIT issue a warning for PI phone numbers with less than 10 digits.

### CODE REVIEW

File: bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/robot/ProblemRobot.scala [hunk 1]
```diff
@@ -612,12 +612,16 @@ class ProblemRobot(s: ShellAdvisor) extends Robot {
...
-        "Overview", s.inOverview{
-          _.editPi(_.Phone.requestFocus)
-        })
+      new Problem(Severity.Warning,
+        s"No phone number given for ${investigatorFullName(p.investigators.pi, "PI")}. This is for improved user support.",
+        "Overview", s.inOverview(_.editPi(_.Phone.requestFocus)))
     }
 
+    private val invalidPIPhoneNumber = (for {
+      badPhone <- p.investigators.pi.phone if badPhone.count(_.isDigit) < 10
```
Comment (cquiroz) [inline bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/robot/ProblemRobot.scala]: Should we also trigger an error if e.g. the phone number contains letters? I guess `+` should be allowed

Comment (sraaphorst) [inline bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/robot/ProblemRobot.scala]: I feel very hesitant to do something like that. Extensions, for example, contain the letter x, typically, and although I don't think there are any other cases in the institutions list now of anything else, it is entirely possible that something might come up. Additionally, how complicated would we want to take this? Parentheses (in which case, do we only allow one set, require them to be balanced, etc), a multitude of separators [.-/], and all kinds of things could crop up, as could an optional leading +, as you mention.

I was trying to find a regex online for phone numbers as a curiosity, and even one for just North America was ridiculously complicated. I don't think we want to take things that far, at least until science requests it.

Comment (cquiroz) [inline bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/robot/ProblemRobot.scala]: That sounds fine, I think you're right

**Comment by cquiroz** [comment]
Looks good to me :+1:

File: bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/robot/ProblemRobot.scala (hunk 1 above)
Comment (swalker2m) [inline bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/robot/ProblemRobot.scala]: I'm even a little surprised that there isn't a country that uses 9 or fewer digits.

Comment (sraaphorst) [inline bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/robot/ProblemRobot.scala]: Actually, looking online, the assumption that a phone number will contain at least 10 digits is wrong. Even with country code, Swedish phone numbers can have nine digits, and phone numbers for the Solomon Islands, eight.

Out of the partner countries, they will all have 10+, but this seems like a poor assumption to make. I'm going to make note on the JIRA task for this.

**Comment by sraaphorst** [comment]
Okay, spoke with @andrewwstephens and we agreed to set it to eight digits, which is the minimum legal number for any international phone number. I also simplified the code to use `find` instead of the ridiculous `for`...`headOption` I was going before.

**Pull request merged**